### PR TITLE
Add mod tag descriptions

### LINKED ISSUE #2439: Add Mod Tag Descriptions
### Do you intend to personally contribute/program this feature?

Maybe
(Note: I can add the descriptions, but I don't know how to make it that way in the code. Someone can add the mod tags internal name in the localization files with the same placeholder descriptions, and I can change the placeholder descriptions into actual descriptions.)

### I would like to see this change made to improve my experience with

Mod capability as a Modder

### Description

There should be actual descriptions of mod tags when selecting mod tags.
Example: "New Content" Tag. Description: "This mod adds new items and/or features to the game".

### What does this proposal attempt to solve or improve?

By removing placeholder descriptions and replacing them with real descriptions, these real descriptions tell you what the mod tags mean, so you can have a better understanding of what mod tags to use.

**Currently:**
image

### Which (other) solutions should be considered?

_No response_

### PR DESCRIPTION
**Pull request by Setnour6**
Also removes the placeholder text thing in other localization files.
If anyone can add translations, that would be great.

### What is the new feature?
Adds actual descriptions of mod tags, so people can know what they are for and if their mod contains content that fall in any of the tags criteria.
### Why should this be part of tModLoader?
There are currently placeholder descriptions. This PR changes that to actual descriptions.
### Are there alternative designs?
_No response_
### Sample usage for the new feature
See Webmilio's Commons Library by Webmilio and Better Mod List by Tomat. These mods implemented mod tags before actual mod tags came into place.

Resolves #2439

### CODE REVIEW

**Comment by Setnour6** [comment]
@Solxanich in case you didn't get notified earlier, this is review ready. I still can't add reviewers for some reason, but I think only tModLoader staff can do so.

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json [hunk 1]
```diff
@@ -438,24 +438,25 @@
...
+		// Tag descriptions by Webmilio, Tomat, and Setnour6.
 		"TagsContent": "New Content",
-		"TagsContentDescription": "This is a placeholder description. Don't Translate!",
+		"TagsContentDescription": "Adds new content (i.e. items, weapons, npcs, etc.) to the game",
 		"TagsUtility": "Utilities",
-		"TagsUtilityDescription": "This is a placeholder description. Don't Translate!",
+		"TagsUtilityDescription": "Adds utility-oriented content or features",
 		"TagsLibrary": "Library",
-		"TagsLibraryDescription": "This is a placeholder description. Don't Translate!",
+		"TagsLibraryDescription": "Utilities or features that other mods use. In other words, Provides useful features for other modders",
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: Would prefer simplified to: 
Library: Provides additional API & features useful to Modders
GameplayTweaks: Tweaks that change or improve vanilla gameplay

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json [hunk 2]
```diff
@@ -438,24 +438,25 @@
...
+		"TagsGameplayDescription": "Adjustments that change existing features (i.e. different UI, etc.)",
 		"TagsAudio": "Audio Tweaks",
-		"TagsAudioDescription": "This is a placeholder description. Don't Translate!",
+		"TagsAudioDescription": "Adjustments that change existing audio (i.e. new boss themes, etc.)",
 		"TagsVisual": "Visual Tweaks",
-		"TagsVisualDescription": "This is a placeholder description. Don't Translate!",
+		"TagsVisualDescription": "Adjustments that change existing visualization (i.e. particle effects, etc.)",
 		"TagsLang": "Localization Support",
-		"TagsLangDescription": "This is a placeholder description. Don't Translate!",
+		"TagsLangDescription": "Adds language support for your mod. In other words, your mod supports different languages",
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: Simplify to
Lang: This mod supports multiple languages.

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json [hunk 3]
```diff
@@ -438,24 +438,25 @@
...
+		"TagsContentDescription": "Adds new content (i.e. items, weapons, npcs, etc.) to the game",
 		"TagsUtility": "Utilities",
-		"TagsUtilityDescription": "This is a placeholder description. Don't Translate!",
+		"TagsUtilityDescription": "Adds utility-oriented content or features",
 		"TagsLibrary": "Library",
-		"TagsLibraryDescription": "This is a placeholder description. Don't Translate!",
+		"TagsLibraryDescription": "Utilities or features that other mods use. In other words, Provides useful features for other modders",
 		"TagsQoL": "Quality Of Life",
-		"TagsQoLDescription": "This is a placeholder description. Don't Translate!",
+		"TagsQoLDescription": "Adjustments that improve the gameplay and/or feel of the game (i.e. improved recipes, etc.)",
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: This one is a little tougher, suggest:
Features or changes that reduce time spent on typically less fun tasks.

**Review by Solxanich — CHANGES_REQUESTED** [review]
A couple of description improvements suggested.

Have you checked these in-game to ensure they aren't too long on the smallest window size? I suspect a few are too wordy

**Comment by Setnour6** [comment]
> Have you checked these in-game to ensure they aren't too long on the smallest window size? I suspect a few are too wordy

I can't check it because I keep getting those child fork process died unexpectedly errors. For some reason, I can't get tModLoader to launch when I debug it with Visual Studio. I'll try again and see if I can get it to work.

**Comment by Setnour6** [comment]
I couldn't get it to work unfortunately. I updated the translations again to reflect the English texts so translators can translate the text easier. I will now re-request the review to check everything again and make sure it all fits well with the description.

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json [hunk 4]
```diff
@@ -438,24 +438,25 @@
...
+		"TagsLibraryDescription": "Provides additional API & features useful to modders",
 		"TagsQoL": "Quality Of Life",
-		"TagsQoLDescription": "This is a placeholder description. Don't Translate!",
+		"TagsQoLDescription": "Features or changes that reduce time spent on typically less fun tasks.",
 		"TagsGameplay": "Gameplay Tweaks",
-		"TagsGameplayDescription": "This is a placeholder description. Don't Translate!",
+		"TagsGameplayDescription": "Tweaks that change or improve vanilla gameplay",
 		"TagsAudio": "Audio Tweaks",
-		"TagsAudioDescription": "This is a placeholder description. Don't Translate!",
+		"TagsAudioDescription": Tweaks that change existing audio (i.e. new boss themes, etc.)",
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: Missing a " at the start of the string.

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json [hunk 5]
```diff
@@ -438,24 +438,25 @@
...
 		"TagsVisual": "Visual Tweaks",
-		"TagsVisualDescription": "This is a placeholder description. Don't Translate!",
+		"TagsVisualDescription": Tweaks that change existing visualization (i.e. particle effects, etc.)",
```
Comment (Solxanich) [inline patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: Missing a " at the start of the string.

File: patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json (hunk 4 above)
Comment (Setnour6) [reply patches/tModLoader/Terraria/Localization/Content/en-US/tModLoader.json]: How did that happen? Oh well, I fixed it.

**Comment by Setnour6** [comment]
I'm still not sure if you get notified by commits, but I think you do, writing this to let you know that I made the necessary fixes. Hopefully everything works well.

**Comment by Solxanich** [comment]
This will serve as a good starting point and can be improved/fine tuned later.
